### PR TITLE
B 18882 int add search capability for too & tio

### PR DESCRIPTION
**Pull request by joeydoyecaci**
## Summary
Added capability for TOO and TIO to be able to search for Moves

### How to test

Prerequisite: Create a move outside of the TOO/TIO GBLOC. Complete move until they populate the queues of TOO/TIO.

### TOO:
1. Login as TOO
2. pick a Move Code, DOD ID, and Name from the first queue for later
3. Click the Search tab
4. Run a search by default (customer name) using tåhe name from earlier, only moves with that name should show
5. Click on status (Verify that these Draft, New Move, Needs Counseling, Service Counseling completed, Move Approved exist)
6. Click on one or any combination to filter down to the move statuses as shown
7. Click on any of the move(s) that have shown after filtering
8. Go back to the search queue
9. You should be able to see the move you created outside of the GBLOC
10. Use the move code from earlier to run another search, verify only moves with that move code show
11. Run the move again using DOD ID as an option, only moves with this ID should show.
12. See: [Testing Move Search Validation](###Testing-Move-Search-Validation) (This only needs to be tested once as this form is shared)
### TIO:
1. Login as TIO
2. pick a Move Code, DOD ID, and Name from the first queue for later
3. Click the Search tab
4. Run a search by default (customer name) using the name from earlier, only moves with that name should show
5. Click on status (Verify that these Draft, New Move, Needs Counseling, Service Counseling completed, Move Approved exist)
6. Click on one or any combination to filter down to the Payment Request statuses as shown
7. Click on any of the move(s) that have shown after filtering
8. Go back to the search queue
9. Use the move code from earlier to run another search, verify only moves with that move code show
10. Run the move again using DOD ID as an option, only moves with this ID should show.
### ADDENDUM 03/28/24 (bug fix added from 18881):
1. Login as Service Counselor
2. Click on the status filter and check that only 'Needs counseling' and 'Service counseling completed' exist as options
3. Select these options at least once, can be in any combination concurrently, but all must have been selected at least once.
4. pick a Move Code, DOD ID, and Name from the first queue for later
5. Click the Search tab
6. Run a search by default (customer name) using the name from earlier, only moves with that name should show
7. Click on status filter dropdown and verify that these Draft, New Move, Needs Counseling, Service counseling completed, and Move Approved exist
8. Click on any of the move(s) that have shown after filtering, its details should show
9. Go back to the search queue
10. Use the move code from earlier to run another search, verify only moves with that move code show
11. Run the move again using DOD ID as an option, only moves with this ID should show.

### Testing Move Search Validation:
1. For searching with move code, verify that an error occurs if a move code < 6 characters or > 6 characters fails validation, including whitespaces
2. Do a search for move code, with a whitespace after, move search should be unaffected
3. Switching between MoveCode,DOD ID, or Customer Name should clear out the text field
4. DOD ID length must == 10. Validation error should occur any other time.
5. Customer name should present a validation error if empty only (or is whitespace).

### CODE REVIEW

File: src/pages/Office/CustomerOnboarding/CreateCustomerForm.jsx [hunk 1]
```diff
@@ -155,7 +155,7 @@ export const CreateCustomerForm = ({ setFlashMessage }) => {
     return createCustomerWithOktaOption({ body })
       .then(() => {
         setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
-        navigate(servicesCounselingRoutes.BASE_QUEUE_SEARCH_PATH);
+        navigate(generalRoutes.BASE_QUEUE_SEARCH_PATH);
```
Comment (danieljordan-caci) [inline src/pages/Office/CustomerOnboarding/CreateCustomerForm.jsx:158]: Nice. Thank you!

**Comment by danieljordan-caci** [comment]
I am not able to get this running locally, likely due to the ampersand in your branch name. Any way you can take that out? I get this error:
```
make client_run
REACT_APP_GIT_COMMIT=7f73d03697de5dd2a470e857959b4a708529ca7f \
        REACT_APP_GIT_BRANCH=B-18882-INT-Add-search-capability-for-TOO-&-TIO \
        HOST=milmovelocal \
        yarn start
/bin/sh: -TIO: command not found
make: *** [client_run] Error 127
```

**Comment by joeydoyecaci** [comment]
Not without rebuilding the entire PR

File: src/pages/Office/MoveQueue/MoveQueue.jsx [hunk 2]
```diff
@@ -158,7 +158,7 @@ const MoveQueue = () => {
         items={[
           <NavLink end className={({ isActive }) => (isActive ? 'usa-current' : '')} to={tooRoutes.BASE_MOVE_QUEUE}>
             <span data-testid="closeout-tab-link" className="tab-title" title="Move Queue">
-              Move Queue
+              Task Order Queue
```
Comment (danieljordan-caci) [inline src/pages/Office/MoveQueue/MoveQueue.jsx:161]: Should this be `Move Task Order Queue` instead of `Task Order`?

**Pull request closed without merging**